src/tpm2_wrap.c

```diff
@@ -409,6 +412,70 @@ int wolfTPM2_ReadPublicKey(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* key,
     return rc;
 }
 
+
+int wolfTPM2_RsaKey_TpmToWolf(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* tpmKey,
+    RsaKey* wolfKey)
+{
+    int rc;
+    byte    e[WOLFTPM2_WRAP_RSA_KEY_BITS / 8];
+    byte    n[WOLFTPM2_WRAP_RSA_KEY_BITS / 8];
+    word32  eSz = sizeof(e);
+    word32  nSz = sizeof(n);
+    word32 exponent;
+
+    if (dev == NULL || tpmKey == NULL || wolfKey == NULL)
+        return BAD_FUNC_ARG;
+
+    XMEMSET(e, 0, sizeof(e));
+    XMEMSET(n, 0, sizeof(n));
+
+    /* load exponenet */
+    exponent = tpmKey->pub.publicArea.parameters.rsaDetail.exponent;
+    if (exponent == 0)
+        exponent = RSA_DEFAULT_PUBLIC_EXPONENT;
+    e[3] = (exponent >> 24) & 0xFF;
+    e[2] = (exponent >> 16) & 0xFF;
+    e[1] = (exponent >> 8)  & 0xFF;
+    e[0] =  exponent        & 0xFF;
+    eSz = 4;
+
+    /* load public key */
+    nSz = tpmKey->pub.publicArea.unique.rsa.size;
+    XMEMCPY(n, tpmKey->pub.publicArea.unique.rsa.buffer, nSz);
+
+    /* load public key portion into wolf RsaKey */
+    rc = wc_RsaPublicKeyDecodeRaw(n, nSz, e, eSz, wolfKey);
+
+    return rc;
+}
+
+int wolfTPM2_RsaKey_WolfToTpm(WOLFTPM2_DEV* dev, RsaKey* wolfKey,
+    WOLFTPM2_KEY* tpmKey)
+{
+    int rc;
+    byte    e[WOLFTPM2_WRAP_RSA_KEY_BITS / 8];
+    byte    n[WOLFTPM2_WRAP_RSA_KEY_BITS / 8];
+    word32  eSz = sizeof(e);
+    word32  nSz = sizeof(n);
+    word32 exponent;
+
+    if (dev == NULL || tpmKey == NULL || wolfKey == NULL)
+        return BAD_FUNC_ARG;
+
+    XMEMSET(e, 0, sizeof(e));
+    XMEMSET(n, 0, sizeof(n));
+
+    /* export the raw public RSA portion */
+    rc = wc_RsaFlattenPublicKey(wolfKey, e, &eSz, n, &nSz);
+    if (rc == 0) {
+        exponent = e[3] << 24 | e[2] << 16 | e[1] << 8 | e[0];
```

Comment:
Has this been tested on keys with e values larger than 3 and other than 65537? I'd recommend following the ato32() pattern already established in wolfcrypt/src/misc.c.